### PR TITLE
fix: bytes serialization issues [APE-1012]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

Issue where Ape models containing Hexbytes would error.

fixes: #1465 
Fixes: APE-1012
Fixes: APE-1016

### How I did it

The main fix was using ethpm_types basemodel instead of a naked pydantic one.
Additionally, switched remaining usages of `hexbytes.Hexbytes` with `ethpm_types.HexBytes`.
Instead of failing validation, blocks now convert hexbytes.

### How to verify it

```python
chain.blocks.head.json()
```

### Checklist

- [x] All changes are completed
- [x] New test cases have been added
- [x] Documentation has been updated

### CODE REVIEW

File: src/ape_ethereum/ecosystem.py [hunk 1]
```diff
@@ -167,6 +168,12 @@ class Block(BlockAPI):
     difficulty: int = 0
     total_difficulty: int = Field(0, alias="totalDifficulty")
 
+    # Type re-declares.
```
Comment (fubuloubu) [inline src/ape_ethereum/ecosystem.py:171]: Is this just to re-arrange the order it serializes as?

**Review by fubuloubu — DISMISSED** [review]

File: src/ape_ethereum/ecosystem.py (hunk 1 above)
Comment (antazoey) [reply src/ape_ethereum/ecosystem.py:171]: no, the `BlockAPI` uses `Any` as the type, but in `ape-ethereum`, we can safely assume it is always going to be HexBytes`.

Comment (antazoey) [reply src/ape_ethereum/ecosystem.py:171]: tbh, i dont remember why they are Any in `BlockAPI`; maybe we can be rid of that

Comment (fubuloubu) [reply src/ape_ethereum/ecosystem.py:171]: I think bytes/HexBytes is probably an appropiate assumption for those fields abstractly

Comment (antazoey) [reply src/ape_ethereum/ecosystem.py:173]: im good with making the change now, shouldnt really break anything except maybe the already broken starknet

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**